src/zenml/integrations/vault/secrets_manager/vault_secrets_manager.py

```diff
@@ -0,0 +1,333 @@
+
+    Returns:
+        The secret name with the secret schema name prepended.
+    """
+    secret_name = sanitize_secret_name(secret.name)
+    secret_schema_name = secret.TYPE
+    return f"{secret_schema_name}-{secret_name}"
+
+
+def remove_secret_schema_name(combined_secret_name: str) -> str:
+    """
+    Remove the secret schema name from the secret name.
+
+    Args:
+        combined_secret_name: The secret name to remove the secret schema name from.
+
+    Returns:
+        The secret name with the secret schema name removed.
+    """
+    if "-" in combined_secret_name:
+        return combined_secret_name.split("-")[1]
+    else:
+        raise RuntimeError(
+            f"Secret name `{combined_secret_name}` does not have a "
+            f"Secret_schema name on it."
+        )
+
+
+def get_secret_schema_name(combined_secret_name: str) -> str:
+    """
+    Get the secret schema name from the secret name.
+
+    Args:
+        combined_secret_name: The secret name to get the secret schema name from.
+
+    Returns:
+        The secret schema name.
+    """
+    if "-" in combined_secret_name:
+        return combined_secret_name.split("-")[0]
+    else:
+        raise RuntimeError(
+            f"Secret name `{combined_secret_name}` does not have a "
+            f"Secret_schema name on it."
+        )
+
+
+class VaultSecretsManager(BaseSecretsManager):
+    """Class to interact with the Vault secrets manager - Key/value Engine.
+
+    Attributes:
+        url: The url of the Vault server.
+        token: The token to use to authenticate with Vault.
+        cert: The path to the certificate to use to authenticate with Vault.
+        verify: Whether to verify the certificate or not.
+        mount_point: The mount point of the secrets manager.
+        namespace: The namespace of the secrets manager.
+    """
+
+    # Class configuration
+    FLAVOR: ClassVar[str] = VAULT_SECRETS_MANAGER_FLAVOR
+    CLIENT: ClassVar[Any] = None
+
+    url: str
+    token: str
+    cert: Optional[str]
+    verify: Optional[str]
+    namespace: Optional[str]
+    mount_point: Optional[str] = "secret"
+
+    @classmethod
+    def _ensure_client_connected(cls, url: str, token: str) -> None:
+
+        if cls.CLIENT is None:
+            # Create a Vault Secrets Manager client
+            cls.CLIENT = hvac.Client(
+                url=url,
+                token=token,
+            )
+
+    def _ensure_client_is_authenticated(self) -> None:
+        """
+        Ensure the client is authenticated.
+
+        Raises:
+            RuntimeError: If the client is not initialized or authenticated.
+        """
+
+        self._ensure_client_connected(url=self.url, token=self.token)
+
+        if not self.CLIENT:
+            raise RuntimeError("Vault client is not initialized.")
+        else:
+            if not self.CLIENT.is_authenticated():
+                raise RuntimeError("Vault client is not authenticated.")
```

Comment:
These two exections don't make it very clear to the user what they can/have to do to fix the issue. Would be awesome if you could help the user troubleshoot here. How can I authenticate. If its simple describe it in the message, if it is a bit harder, just supply the relevant hvac docs.